jib-core/src/main/java/com/google/cloud/tools/jib/event/progress/AllocationCompletionTracker.java

```diff
@@ -107,7 +107,9 @@ boolean updateProgress(Allocation allocation, long units) {
 
   /**
    * Gets a list of the unfinished {@link Allocation}s in the order in which those {@link
-   * Allocation}s were encountered.
+   * Allocation}s were encountered. This can be used to display, for example, currently executing
+   * tasks. The order helps to keep the displayed tasks in a deterministic order (new subtasks
```

Comment:
> new subtasks appear below older ones

I personally think it doesn't really matter if a new subtask appear above older ones, as long as the displayer does not dynamically change the order once it displayed some tasks. Not having to think about the order here will make the code simpler, and the displayer just needs to manage the order consistently on its side. What do you think?

Reply:
One of the things I've noticed in the prototype is that since a normal map doesn't give a deterministic order, when a new subtask is added, it could change the relative orders of tasks that existed before (when getting the unfinished tasks in some order from the map). This makes it sometimes hard to keep track of which is the new task that started vs an old task that is still running. We could manage the order from the displayer side but that would mean that the displayer would need to keep some kind of map/association from allocation to an order index and sort all unfinished ones by the order index, which results in concurrent code that would basically be the same. I think we can explore moving the order tracking to the display side once we add that and see which way is cleaner?

Reply:
> which results in concurrent code that would basically be the same.

Yeah, I can imagine that too. Let's move forward with managing the order here.